magento/pwa-studio
# Fix Service Worker matching external top-level URLs

## Description

At the moment the Service Worker matches any external top-level path (e.g. https://any.external.domain.com/?query_params) as a home url, as `isHomeRoute` makes a mere string equality check against `url.pathname`. This makes it return the index template for any such request.

I stumbled upon these while trying to load the script for Zendesk Chat, whose URL has the form `https://v2.zopim.com/?APP_KEY_HASH`.

This commit introduces an additional check to make sure that the URL origin matches the service worker origin, too, just as it would happen when passing a `RegExp` (bc of additional assumptions of workbox on the use of `RegExp`s that only make them match on local URLS)

## Related Issue
* Closes #3193

## Acceptance



### Verification Stakeholders


### Verification Steps
* Run `fetch('https://v2.zopim.com/?foo',{mode: 'no-cors'}).then(data => console.log(data))` in the console
* Verify that the service work doesn't intercept the request

### Specification





### Resolved issues:
1. [x] resolves magento/pwa-studio#3193: Fix Service Worker matching external top-level URLs
